src/utils.py

```diff
@@ -159,45 +159,69 @@ def create_package_dict(graph_results, alt_dict=None):
 def convert_version_to_proper_semantic(version):
     """Perform Semantic versioning.
 
-    :param version: The raw input version that needs to be converted.
-    :return: The semantic version of raw input version.
-
-    Needed for maven version like 1.5.2.RELEASE to be converted to
-    1.5.2-RELEASE for semantic version to work'
+    : type version: string
+    : param version: The raw input version that needs to be converted.
+    : type return: semantic_version.base.Version
+    : return: The semantic version of raw input version.
     """
     if version in ('', '-1', None):
-        return '0.0.0'
+        version = '0.0.0'
+    """Needed for maven version like 1.5.2.RELEASE to be converted to
+    1.5.2 - RELEASE for semantic version to work."""
     version = version.replace('.', '-', 3)
     version = version.replace('-', '.', 2)
-    return version
+    # Needed to add this so that -RELEASE is account as a Version.build
+    version = version.replace('-', '+', 3)
+    return sv.Version.coerce(version)
 
 
-def select_latest_version(input_version='', libio='', anitya=''):
-    """Select latest version from input sequence(s)."""
-    libio_latest_version = convert_version_to_proper_semantic(libio)
-    anitya_latest_version = convert_version_to_proper_semantic(anitya)
-    input_version = convert_version_to_proper_semantic(input_version)
+def version_info_tuple(version):
+    """Return the sem_version information in form of (major, minor, patch, build).
 
-    try:
-        return_version = input_version
-        if sv.Version(libio_latest_version) >= sv.Version(anitya_latest_version)\
-                and sv.Version(libio_latest_version) >= sv.Version(input_version):
-            return_version = libio
+    : type version: semantic_version.base.Version
+    : param version: The semantic version whole details are needed.
+    : return: A tuple in form of Version.(major, minor, patch, build)
+    """
+    if type(version) == sv.base.Version:
```

Comment:
I think you could use [`isinstance`](https://docs.python.org/3/library/functions.html#isinstance) here. It would also cover cases when version is a subclass of `sv.base.Version`.